src/image/imageloadobserver.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module image/image/imageloadobserver
+ */
+
+import DomEventObserver from '@ckeditor/ckeditor5-engine/src/view/observer/domeventobserver';
+import DOMEmitterMixin from '@ckeditor/ckeditor5-utils/src/dom/emittermixin';
+
+/**
+ * Observes all new images added to the {@link module:engine/view/document~Document},
+ * fires {@link module:engine/view/document~Document#event:imageLoaded} and
+ * {@link module:engine/view/document~Document#layoutChanged} event every time when the new image
+ * has been loaded.
+ *
+ * **Note:** This event is not fired for images that has been added to the document and rendered as `complete` (already loaded).
+ *
+ * @extends module:engine/view/observer/domeventobserver~DomEventObserver
+ */
+export default class ImageLoadObserver extends DomEventObserver {
```

Comment:
Not sure if it should be `DomEventObserver`. You overwrite most of it anyway and the only part you do not overwrite, `fire` method is misleading.